File: win/directx/sdk-9.0/Samples/C#/DirectPlay/Tutorials/Tut04_Connect/HostDialog.cs
//----------------------------------------------------------------------------
// File: HostDialog.cs
//
// Copyright (c) Microsoft Corp. All rights reserved.
//-----------------------------------------------------------------------------
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace Tut04_Connect
{
    /// <summary>
    /// Summary description for HostDialog.
    /// </summary>
    public class HostDialog : System.Windows.Forms.Form
    {
        private string   m_SessionName           = "New Host";  // Hosted session name 
       
        private System.Windows.Forms.Label label1;
        private System.Windows.Forms.Label label2;
        private System.Windows.Forms.PictureBox pictureBox1;
        private System.Windows.Forms.Button OKButton;
        private System.Windows.Forms.Button cancelButton;
        private System.Windows.Forms.TextBox SessionNameTextBox;
        private System.Windows.Forms.Label SessionNameLabel;
        private System.Windows.Forms.TextBox portTextBox;
        private System.Windows.Forms.Label portLabel;

        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components = null;

        // Properties
        public string SessionName{ get{ return m_SessionName; } } 
        
        /// <summary>
        /// Property: Local port on which to host
        /// </summary>
        public int LocalPort
        {
            get
            { 
                int retValue = 0;

                try
                {
                    retValue = int.Parse(portTextBox.Text); 
                }
                catch (Exception) {}
                
                return retValue;
            }
            set
            { portTextBox.Text = value.ToString(); }
        }

        /// <summary>
        /// Constructor
        /// </summary>
        public HostDialog()
        {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
            SessionNameTextBox.Text = m_SessionName;
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

		#region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.pictureBox1 = new System.Windows.Forms.PictureBox();
            this.OKButton = new System.Windows.Forms.Button();
            this.cancelButton = new System.Windows.Forms.Button();
            this.SessionNameTextBox = new System.Windows.Forms.TextBox();
            this.SessionNameLabel = new System.Windows.Forms.Label();
            this.portTextBox = new System.Windows.Forms.TextBox();
            this.portLabel = new System.Windows.Forms.Label();
            this.SuspendLayout();
            // 
            // label1
            // 
            this.label1.BackColor = System.Drawing.SystemColors.Info;
            this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
            this.label1.Location = new System.Drawing.Point(15, 16);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(224, 16);
            this.label1.TabIndex = 0;
            this.label1.Text = "Please provide a name for the new session.";
            // 
            // label2
            // 
            this.label2.BackColor = System.Drawing.SystemColors.Info;
            this.label2.Location = new System.Drawing.Point(15, 32);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(232, 16);
            this.label2.TabIndex = 1;
            this.label2.Text = "This name will help other players identify you.";
            // 
            // pictureBox1
            // 
            this.pictureBox1.BackColor = System.Drawing.SystemColors.Info;
            this.pictureBox1.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
            this.pictureBox1.Location = new System.Drawing.Point(8, 8);
            this.pictureBox1.Name = "pictureBox1";
            this.pictureBox1.Size = new System.Drawing.Size(240, 48);
            this.pictureBox1.TabIndex = 2;
            this.pictureBox1.TabStop = false;
            // 
            // OKButton
            // 
            this.OKButton.Location = new System.Drawing.Point(112, 144);
            this.OKButton.Name = "OKButton";
            this.OKButton.Size = new System.Drawing.Size(64, 24);
            this.OKButton.TabIndex = 3;
            this.OKButton.Text = "&OK";
            this.OKButton.Click += new System.EventHandler(this.OKButton_Click);
            // 
            // cancelButton
            // 
            this.cancelButton.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.cancelButton.Location = new System.Drawing.Point(184, 144);
            this.cancelButton.Name = "cancelButton";
            this.cancelButton.Size = new System.Drawing.Size(64, 24);
            this.cancelButton.TabIndex = 4;
            this.cancelButton.Text = "&Cancel";
            this.cancelButton.Click += new System.EventHandler(this.cancelButton_Click);
            // 
            // SessionNameTextBox
            // 
            this.SessionNameTextBox.AutoSize = false;
            this.SessionNameTextBox.Location = new System.Drawing.Point(104, 72);
            this.SessionNameTextBox.Name = "SessionNameTextBox";
            this.SessionNameTextBox.Size = new System.Drawing.Size(128, 20);
            this.SessionNameTextBox.TabIndex = 5;
            this.SessionNameTextBox.Text = "New Host";
            // 
            // SessionNameLabel
            // 
            this.SessionNameLabel.Location = new System.Drawing.Point(16, 74);
            this.SessionNameLabel.Name = "SessionNameLabel";
            this.SessionNameLabel.Size = new System.Drawing.Size(83, 16);
            this.SessionNameLabel.TabIndex = 6;
            this.SessionNameLabel.Text = " Session Name ";
            this.SessionNameLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // portTextBox
            // 
            this.portTextBox.Location = new System.Drawing.Point(104, 104);
            this.portTextBox.Name = "portTextBox";
            this.portTextBox.Size = new System.Drawing.Size(56, 20);
            this.portTextBox.TabIndex = 7;
            this.portTextBox.Text = "";
            // 
            // portLabel
            // 
            this.portLabel.Location = new System.Drawing.Point(48, 106);
            this.portLabel.Name = "portLabel";
            this.portLabel.Size = new System.Drawing.Size(48, 16);
            this.portLabel.TabIndex = 8;
            this.portLabel.Text = " Port ";
            this.portLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // HostDialog
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.CancelButton = this.cancelButton;
            this.ClientSize = new System.Drawing.Size(258, 176);
            this.ControlBox = false;
            this.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                          this.portLabel,
                                                                          this.portTextBox,
                                                                          this.SessionNameLabel,
                                                                          this.SessionNameTextBox,
                                                                          this.cancelButton,
                                                                          this.OKButton,
                                                                          this.label2,
                                                                          this.label1,
                                                                          this.pictureBox1});
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "HostDialog";
            this.ShowInTaskbar = false;
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Host New Session";
            this.ResumeLayout(false);

        }
		#endregion

        private void OKButton_Click(object sender, System.EventArgs e)
        {
            // Save the current settings
            m_SessionName = SessionNameTextBox.Text;
            
            DialogResult = DialogResult.OK;
        }

        private void cancelButton_Click(object sender, System.EventArgs e)
        {
            // Discard settings
            DialogResult = DialogResult.Cancel;
        }
    }
}

File: win/directx/sdk-9.0/Samples/VB.Net/DirectPlay/Tutorials/Tut04_Connect/HostDialog.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Tut04_Connect

    '/ <summary>
    '/ Summary description for HostDialog.
    '/ </summary>
    Public Class HostDialog
        Inherits System.Windows.Forms.Form
        Private m_SessionName As String = "New Host" ' Hosted session name 
        Private label1 As System.Windows.Forms.Label
        Private label2 As System.Windows.Forms.Label
        Private pictureBox1 As System.Windows.Forms.PictureBox
        Private WithEvents OKButton As System.Windows.Forms.Button
        Private WithEvents buttonCancel As System.Windows.Forms.Button
        Private SessionNameTextBox As System.Windows.Forms.TextBox
        Private SessionNameLabel As System.Windows.Forms.Label

        '/ <summary>
        '/ Required designer variable.
        '/ </summary>
        Private components As System.ComponentModel.Container = Nothing

        ' Properties

        Public ReadOnly Property SessionName() As String
            Get
                Return m_SessionName
            End Get
        End Property

        ' Property: Local port on which to host
        Public Property LocalPort() As Integer
            Get
                Dim retValue As Integer

                Try
                    retValue = Integer.Parse(portTextBox.Text)
                Catch ex As Exception
                End Try

                Return retValue
            End Get

            Set(ByVal Value As Integer)
                portTextBox.Text = Value.ToString()
            End Set
        End Property

        '/ <summary>
        '/ Constructor
        '/ </summary>
        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()
            SessionNameTextBox.Text = m_SessionName
        End Sub 'New

        '/ <summary>
        '/ Clean up any resources being used.
        '/ </summary>
        Protected Overloads Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub 'Dispose

        '/ <summary>
        '/ Required method for Designer support - do not modify
        '/ the contents of this method with the code editor.
        '/ </summary>
        Friend WithEvents portLabel As System.Windows.Forms.Label
        Friend WithEvents portTextBox As System.Windows.Forms.TextBox
        Private Sub InitializeComponent()
            Me.label1 = New System.Windows.Forms.Label()
            Me.label2 = New System.Windows.Forms.Label()
            Me.pictureBox1 = New System.Windows.Forms.PictureBox()
            Me.OKButton = New System.Windows.Forms.Button()
            Me.buttonCancel = New System.Windows.Forms.Button()
            Me.SessionNameTextBox = New System.Windows.Forms.TextBox()
            Me.SessionNameLabel = New System.Windows.Forms.Label()
            Me.portLabel = New System.Windows.Forms.Label()
            Me.portTextBox = New System.Windows.Forms.TextBox()
            Me.SuspendLayout()
            '
            'label1
            '
            Me.label1.BackColor = System.Drawing.SystemColors.Info
            Me.label1.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
            Me.label1.Location = New System.Drawing.Point(15, 16)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(224, 16)
            Me.label1.TabIndex = 0
            Me.label1.Text = "Please provide a name for the new session."
            '
            'label2
            '
            Me.label2.BackColor = System.Drawing.SystemColors.Info
            Me.label2.Location = New System.Drawing.Point(15, 32)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(232, 16)
            Me.label2.TabIndex = 1
            Me.label2.Text = "This name will help other players identify you."
            '
            'pictureBox1
            '
            Me.pictureBox1.BackColor = System.Drawing.SystemColors.Info
            Me.pictureBox1.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
            Me.pictureBox1.Location = New System.Drawing.Point(8, 8)
            Me.pictureBox1.Name = "pictureBox1"
            Me.pictureBox1.Size = New System.Drawing.Size(240, 48)
            Me.pictureBox1.TabIndex = 2
            Me.pictureBox1.TabStop = False
            '
            'OKButton
            '
            Me.OKButton.Location = New System.Drawing.Point(112, 160)
            Me.OKButton.Name = "OKButton"
            Me.OKButton.Size = New System.Drawing.Size(64, 24)
            Me.OKButton.TabIndex = 3
            Me.OKButton.Text = "&OK"
            '
            'buttonCancel
            '
            Me.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
            Me.buttonCancel.Location = New System.Drawing.Point(184, 160)
            Me.buttonCancel.Name = "buttonCancel"
            Me.buttonCancel.Size = New System.Drawing.Size(64, 24)
            Me.buttonCancel.TabIndex = 4
            Me.buttonCancel.Text = "&Cancel"
            '
            'SessionNameTextBox
            '
            Me.SessionNameTextBox.AutoSize = False
            Me.SessionNameTextBox.Location = New System.Drawing.Point(104, 80)
            Me.SessionNameTextBox.Name = "SessionNameTextBox"
            Me.SessionNameTextBox.Size = New System.Drawing.Size(120, 20)
            Me.SessionNameTextBox.TabIndex = 5
            Me.SessionNameTextBox.Text = "New Host"
            '
            'SessionNameLabel
            '
            Me.SessionNameLabel.Location = New System.Drawing.Point(16, 82)
            Me.SessionNameLabel.Name = "SessionNameLabel"
            Me.SessionNameLabel.Size = New System.Drawing.Size(88, 14)
            Me.SessionNameLabel.TabIndex = 6
            Me.SessionNameLabel.Text = "Session Name"
            Me.SessionNameLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight
            '
            'portLabel
            '
            Me.portLabel.Location = New System.Drawing.Point(51, 114)
            Me.portLabel.Name = "portLabel"
            Me.portLabel.Size = New System.Drawing.Size(48, 16)
            Me.portLabel.TabIndex = 7
            Me.portLabel.Text = "Port"
            Me.portLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight
            '
            'portTextBox
            '
            Me.portTextBox.Location = New System.Drawing.Point(104, 112)
            Me.portTextBox.Name = "portTextBox"
            Me.portTextBox.Size = New System.Drawing.Size(56, 20)
            Me.portTextBox.TabIndex = 8
            Me.portTextBox.Text = ""
            '
            'HostDialog
            '
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(258, 192)
            Me.ControlBox = False
            Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.portTextBox, Me.portLabel, Me.SessionNameLabel, Me.SessionNameTextBox, Me.buttonCancel, Me.OKButton, Me.label2, Me.label1, Me.pictureBox1})
            Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "HostDialog"
            Me.ShowInTaskbar = False
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
            Me.Text = "Host New Session"
            Me.ResumeLayout(False)

        End Sub 'InitializeComponent

        Private Sub OKButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles OKButton.Click '
            ' Save the current settings
            m_SessionName = SessionNameTextBox.Text

            DialogResult = DialogResult.OK
        End Sub 'OKButton_Click


        Private Sub cancelButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonCancel.Click
            ' Discard settings
            DialogResult = DialogResult.Cancel
        End Sub 'cancelButton_Click
    End Class 'HostDialog
End Namespace 'Tut04_Connect